components/service-mgt/axis2-service-mgt/org.wso2.carbon.service.mgt.ui/src/main/resources/web/service-mgt/index.jsp

```diff
@@ -501,15 +508,22 @@ padding:0 10px;
             <% } %>
             <% if (service.getServiceType().equalsIgnoreCase("proxy")) { %>
             <% hasProxy = true; %>
+            <% if ("proxy".equalsIgnoreCase(service.getServiceType())) {
+               String cApp_edit = "../" + service.getServiceType() + "/editCAppartifact.jsp?serviceName=" + serviceName;
+             %>
+                <jsp:include page="<%= cApp_edit%>"/>
+            <% } else { %>
             <td>
                 <a title="Edit '<%=service.getName()%>' in the design view" href="#" onclick="editPS('<%=service.getName()%>');return false;">
                     <img src="../proxyservices/images/design-view.gif" alt="" border="0"> Design View</a>
             </td>
             <td>
-                <a title="Edit '<%=service.getName()%>' in the source view editor" 
-                    style="background-image: url(../proxyservices/images/source-view.gif);" 
-                    class="icon-link" onclick="editProxySourceView('<%=service.getName()%>')" href="#">Source View</a>
+                <a title="Edit '<%=service.getName()%>' in the source view editor"
+                   style="background-image: url(../proxyservices/images/source-view.gif);"
```

Comment:
Why is there a link to proxyservices here? carbon-deployment should not have any references to that.